nkms/client.py

```diff
@@ -55,13 +77,35 @@ def encrypt_key(self, key, pubkey=None, path=None, algorithm=None):
         :return: Encrypted key(s)
         :rtype: bytes
         """
-        pass
+        if not pubkey:
+            pubkey = self._pub_key
 
-    def decrypt_key(self, key, pubkey=None, path=None, owner=None):
+        if len(path) > 1 and not type(path) is str:
+            enc_keys = []
+            for subpath in path:
+                path_pubkey = self._derive_path_key(subpath)
+                enc_keys.append(self.encrypt_key(key, pubkey=path_pubkey))
+            return enc_keys
+        elif type(path) is str:
+            return self.encrypt_key(key, pubkey=self._derive_path_key(path))
```

Comment:
Ideally, I'd like to split the path into subpaths here. Or, better, call a function which does it

Reply:
So, as we discussed earlier, you'd like it so that we pass `path` as a string and then we split it into the paths ourselves?

In pseudocode:

```
path = '/foo/bar'
subpaths = path_split(path)
```
Where `subpaths` is a tuple or a list that has each path?

Reply:
Yeah, I wanted this to be a "high-level" function where splitting into subpaths happens inside

Reply:
Understood, will do.